### PR TITLE
feat: added support for multiple primary key attributes

### LINKED ISSUE #11392: Add types for primaryKeyAttributes into Model class
Please add `primaryKeyAttributes` to Model class to allow access to primaryKeyAttributes from TypeScript without errors.

File: sequelize/types/lib/model.d.ts
```typescript
export abstract class Model<T = any, T2 = any> extends Hooks {
  /** The name of the database table */
  public static readonly tableName: string;

  /**
   * The name of the primary key attribute
   */
  public static readonly primaryKeyAttribute: string;

  /**
   * The name of the primary key attributes
   */
  public static readonly primaryKeyAttributes: string[];
```

### PR DESCRIPTION
**Pull request by nainkunal933**
added support for multiple primary key attributes in sequelize/types/lib/model.d.ts

closes #11392

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by papb** [comment]
Hello! Thank you for the PR!

Please add a test for the typing you added. If you don't know how to do it, please take a look on other recently merged PRs with the `type: typescript` label. Feel free to ping me if you have questions.

**Comment by nainkunal933** [comment]
Hi @papb, I was going through the test cases in the previous PR. I am planning to add the test case in ```types/test/model.ts```. The current model has only one index 'foo'. How do I add another index in the array? I do not know the other fields of the table.
Current Model: 
Snip20190909_10
 I am trying to do the following:
```typescript
indexes: [
    {
      fields: ['foo'],
      using: 'gin',
      operator: 'jsonb_path_ops',
    },
   {
      fields: ['COLUMN_NAME_HERE'],
      using: 'gin',
      operator: 'jsonb_path_ops',
    }
  ],
```
If this works, then it will test whether the DB will accept composite primary keys with multiple columns. Please let me know if I am on the right track. Thank you.

**Comment by papb** [comment]
It's simpler than this, you just have to add some code that compiles with your PR and wouldn't compile without it.

**Comment by nainkunal933** [comment]
Hi @papb, I have added the test case in mode.ts. Please let me know if it is what you expected. 

Sorry for my ignorance. I am new to open source. Thank you for helping me.

**Comment by nainkunal933** [comment]
Also, in terms of documentation, the [model definition page](https://sequelize.org/master/manual/models-definition.html#indexes) has documentation for multiple indexes. So, for now, I will consider the documentation update to be done.
Snip20190909_11

**Comment by nainkunal933** [comment]
Hi @papb, any updates on this PR.

**Comment by papb** [comment]
Hi, thanks for the heads up, somehow your PR got lost among my notifications.

I think you are a bit confused on the intent your own PR. Recall that this is merely an issue with the typescript typings provided by sequelize, which means that the feature itself already exist and works, the only issue is that when someone tries to do `MyModel.primaryKeyAttributes`, the typescript compiler will complain, since it doesn't know that models have a `primaryKeyAttributes` field. The word 'test' might have confused you here (and surely the contributing guidelines deserve an update, I am working on that already by the way!), but a 'test' in the context of typescript typings is just a code that compiles. What it does when executed is irrelevant.

So, in summary, your test is too complicated. Here is a super mini test that would do the job:

```ts
class TestModel extends Model {};
TestModel.primaryKeyAttributes;
```

The above code compiles with your PR and does not compile without it, and it is very short, which is great. Do you see that your test was unnecessarily complicated? Please update your PR with this simpler code.

**Comment by papb** [comment]
> Also, in terms of documentation, the model definition page has documentation for multiple indexes. So, for now, I will consider the documentation update to be done.

Although it is true that this PR does not need any documentation update (since it is just a typings update), again I see that you are a bit lost (don't worry!) - what you've posted about indexes is not related to the task at hand. You're confusing indexes with foreign keys; indexes are another thing.

File: types/test/model.ts [hunk 1]
```diff
@@ -67,3 +67,19 @@ UserModel.findCreateFind({
...
+  description: DataTypes.TEXT,
+}, { sequelize, modelName: 'ModelCompositePrimaryKey' });
+
+ModelCompositePrimaryKey
+  .create({ primaryKeyOne: "Type 1", title: 'Test Record', description: 'Record created to test composite primary key' })
+  .then(modelCompositePrimaryKey => {
+    console.log(modelCompositePrimaryKey.get('title'));
+    console.log(modelCompositePrimaryKey.get('description'));
+    console.log(ModelCompositePrimaryKey.primaryKeyAttributes);
+  })
```
Comment (papb) [inline types/test/model.ts]: As I said above, please simplify all this to

```ts
class TestModel extends Model {};
TestModel.primaryKeyAttributes;
```

Comment (nainkunal933) [reply types/test/model.ts]: Hi @papb, I understand my test was too complicated. Thank you for the thorough explanation. I have made the changes. Please let me know if the changes are ok.
